plat/imx/common/imx_sip_handler.c

```diff
@@ -188,6 +196,49 @@ int imx_src_handler(uint32_t smc_fid,
 	case IMX_SIP_SRC_IS_M_CORE_STARTED:
 		val = mmio_read_32(IMX_IOMUX_GPR_BASE + IOMUXC_GPR22);
 		return !(val & GPR_M7_CPUWAIT);
+	case IMX_SIP_SRC_STOP_M_CORE:
+		/*
+		 * Safe stop
+		 *    If M4 already in WFI,  perform below steps.
+		 * a)	Set [0x303A_002C].0=0   [ request SLEEPHOLDREQn ]
+		 * b)	Wait  [0x303A_00EC].1 = 0  [ wait SLEEPHOLDACKn ]
+		 * c)	Set  GPR.CPUWAIT=1
+		 * d)	Set [0x303A_002C].0=1  [ de-assert SLEEPHOLDREQn ]
+		 * e)	Set SRC_M7_RCR[3:0] = 0xE0   [ reset M7 core/plat ]
+		 * f)	Wait SRC_M7_RCR[3:0] = 0x8
+		 * The following steps move to start part.
+		 * g/h is actually no needed here.
+		 * g)	Init TCM or DDR
+		 * h)	Set GPR.INITVTOR
+		 * i)	Set GPR.CPUWAIT=0,  M7 starting running
+		 */
+		offset = LPS_CPU1;
+		val = mmio_read_32(IMX_GPC_BASE + offset);
+		/* Not in stop/wait mode */
+		if (!(val & (0x3 << 24))) {
```

Comment:
Please use the `GENMASK` here and define bits macros

Reply:
@Qingwu-Li

What I really meant here is that the `GENMASK()` shall be put in the header files with some meaningful names for those bit areas.

Since they are not officially labeled in any Reference Manual, we can come up with any name that indicates their usage.

Then you can use those new macros in the code.

Does that make sense?

Reply:
Exactly, checked all the imx8 Reference Manual, none of them specified it, I named those bits according the nxp comments.